File: network/Neuron.py
from network.Matrix import Matrix
from network import constants
import random

class Neuron:

    def __init__(self, func, activation = None):
        self.func = func
        self.activation = activation
        self.previousInput = None
        self.prevResult = 0

    def process(self, input):
        self.previousInput = input
        result = self.callFunc(input)
        if self.activation != None:
            result = self.activation(result)
        
        self.prevResult = result
        return result

    def adjustWeights(self, deltas):
        pass

    def callFunc(self, input):
        return self.func(input)

File: java/src/Neuron.java
import java.util.*;
import java.util.concurrent.Callable;

class Neuron {

    protected Callable<Object> func;
    protected Callable<Object> activation;
    protected Object previousInput;
    protected int numForwardNeurons;
    protected final Object[] prevResult;
    protected Object currentValue;

    public Neuron(Callable<Object> func) {
        this(func, null);
    }

    public Neuron(Callable<Object> func, Callable<Object> activation) {
        this(func, activation, 1);
    }

    public Neuron(Callable<Object> func, Callable<Object> activation, int numForwardNeurons) {
        this.func = func;
        this.activation = activation;
        this.previousInput = null;
        this.numForwardNeurons = numForwardNeurons
        this.prevResult = new Object[numForwardNeurons];
        this.currentValue = null;
    }

    public Object process(Object input) {
        this.previousInput = input;

        Object value = this.callFunc(input);
        if (this.activation != null) {
            value = this.activation.call()(value);
        }

        this.currentValue = value;
        return value;
    }

    protected void adjustWeights(List<Double> deltas) {

    }

    protected Object callFunc(Object input) {
        return this.func.call(input);
    }
}